src/net_processing.cpp

```diff
@@ -1267,52 +1285,92 @@ void static ProcessGetBlockData(CNode* pfrom, const CInv& inv, CConnman* connman
     }
 }
 
+//! Determine whether or not a peer can request a transaction, and return it (or nullptr if not found or not allowed).
+CTransactionRef static FindTxForGetData(const CNode* peer, const uint256& txid, const std::chrono::seconds mempool_req, const std::chrono::seconds now)
+{
+    auto txinfo = mempool.info(txid);
+    if (txinfo.tx) {
+        // If a TX could have been INVed in reply to a MEMPOOL request,
+        // or is older than UNCONDITIONAL_RELAY_DELAY, permit the request
+        // unconditionally.
+        if ((mempool_req.count() && txinfo.nTime <= mempool_req.count()) || txinfo.nTime <= (now - UNCONDITIONAL_RELAY_DELAY).count()) {
```

Comment:
It is because in `mempool.info()` we use `int64_t` for time whereas in bitcoin its `std::chrono::seconds`. So c++ compiler complains about the different types in the comparison operation.